app/controlplane/plugins/core/guac/v1/guac.go

```diff
@@ -0,0 +1,268 @@
+	}
+
+	// Check that the credentials are valid and the bucket exists
+	client, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(request.Credentials)))
+	if err != nil {
+		return nil, fmt.Errorf("creating storage client: %w", err)
+	}
+
+	_, err = loadBucket(ctx, client, request.Bucket)
+	if err != nil {
+		return nil, err
+	}
+
+	// Store the bucket and provider in the state
+	rawConfig, err := sdk.ToConfig(&registrationState{
+		Bucket:   request.Bucket,
+		Provider: string(providerGCS),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("marshalling configuration: %w", err)
+	}
+
+	return &sdk.RegistrationResponse{
+		// We also want to store the credentials securely
+		Credentials:   &sdk.Credentials{Password: request.Credentials},
+		Configuration: rawConfig,
+	}, nil
+}
+
+// Attachment is executed when to attach a registered instance of this integration to a specific workflow
+func (i *Integration) Attach(_ context.Context, _ *sdk.AttachmentRequest) (*sdk.AttachmentResponse, error) {
+	// Nothing to do during attachment
+	return &sdk.AttachmentResponse{}, nil
+}
+
+const filePrefix = "chainloop"
+
+// Execute will be instantiate when either an attestation or a material has been received
+// It's up to the plugin builder to differentiate between inputs
+func (i *Integration) Execute(ctx context.Context, req *sdk.ExecutionRequest) error {
+	i.Logger.Info("execution requested")
+
+	// Extract registration and attachment configuration if needed
+	var registrationConfig *registrationState
+	if err := sdk.FromConfig(req.RegistrationInfo.Configuration, &registrationConfig); err != nil {
+		return fmt.Errorf("invalid registration configuration %w", err)
+	}
+
+	client, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(req.RegistrationInfo.Credentials.Password)))
+	if err != nil {
+		return fmt.Errorf("creating storage client: %w", err)
+	}
+
+	bucket, err := loadBucket(ctx, client, registrationConfig.Bucket)
+	if err != nil {
+		return fmt.Errorf("loading bucket: %w", err)
+	}
+
+	// 1 - Upload the attestation
+	envelopeJSON, err := json.Marshal(req.Input.Attestation.Envelope)
+	if err != nil {
+		return fmt.Errorf("marshalling attestation: %w", err)
+	}
+
+	// chainloop-deadbeef-attestation.json
+	fileName := fmt.Sprintf("%s-%s-attestation.json", filePrefix, req.Input.Attestation.Hash.Hex)
+	if err := uploadToBucket(ctx, bucket, fileName, envelopeJSON, req.ChainloopMetadata, i.Logger); err != nil {
+		return fmt.Errorf("uploading the SBOM to the bucket: %w", err)
+	}
+
+	// 2 - Upload all the materials, in our case they are SBOMs
+	for _, sbom := range req.Input.Materials {
+		// Validate that in fact they are SBOMs
+		if err := validateExecuteOpts(sbom, req.RegistrationInfo, req.AttachmentInfo); err != nil {
+			i.Logger.Warnw("msg", "invalid material received", "err", err)
+			continue
+		}
+
+		// chainloop-deadbeef-sbom.json
+		fileName := fmt.Sprintf("%s-%s-%s", filePrefix, sbom.Hash.Hex, sbom.Value)
```

Comment:
what if we do something like that here:
-  `fmt.Sprintf("%s-%s-%s", filePrefix, sbom.Value, sbom.Hash.Hex)`
- or `fmt.Sprintf("%s-%s-%s.%s", filePrefix, base(sbom.Value), sbom.Hash.Hex, rxt(sbom.Value))`

Reply:
> fmt.Sprintf("%s-%s-%s", filePrefix, sbom.Value, sbom.Hash.Hex)

It will make it look like `chainloop-sbom.json-deadbeef)`

Reply:
> or fmt.Sprintf("%s-%s-%s.%s", filePrefix, base(sbom.Value), sbom.Hash.Hex, rxt(sbom.Value))

If I understand correctly, what you are implicitly saying is that you'd prefer a different format on the filename that includes the original filename at the front, followed by its digest correct?

And yes, I'll take a look at it.

Reply:
I am not sure anymore :) Any naming conventions people follow?
```
chainloop-deadbadadadasdasdasdadasdeef-sbom.json
chainloop-sbom-deadbadadadasdasdasdadasdeef.json
chainloop/sbom.json/deadbadadadasdasdasdadasdeef
chainloop-sbom.json-deadbadadadasdasdasdadasdeef
```

Reply:
Updated, ptal 

![image](https://github.com/chainloop-dev/chainloop/assets/24523/ed4675d3-3795-4e02-81eb-a98bfd66ecc8)